py/desiutil/bitmask.py

```diff
@@ -0,0 +1,168 @@
+            bitdefs : dictionary of different mask bit definitions;
+                each value is a list of [bitname, bitnum, comment]
+
+        Typical users are not expected to create BitMask objects directly.
+        """
+        self._bits = dict()
+        self._name = name
+        for x in bitdefs[name]:
+            bitname, bitnum, comment = x[0:3]
+            if len(x) == 4:
+                extra = x[3]
+            else:
+                extra = dict()
+            self._bits[bitname] = _MaskBit(bitname, bitnum, comment, extra)
+            self._bits[bitnum] = self._bits[bitname]
+
+    def __getitem__(self, bitname):
+        return self._bits[bitname]
+
+    def bitnum(self, bitname):
+        """Return bit number (int) for bitname (string)"""
+        return self._bits[bitname].bitnum
+
+    def bitname(self, bitnum):
+        """Return bit name (string) for this bitnum (integer)"""
+        return self._bits[bitnum].name
+
+    def comment(self, bitname_or_num):
+        """Return comment for this bit name or bit number"""
+        return self._bits[bitname_or_num].comment
+
+    def mask(self, name_or_num):
+        """Return mask value, e.g.
+
+        bitmask.mask(3)         #  2**3
+        bitmask.mask('BLAT')
+        bitmask.mask('BLAT|FOO')
+        """
+        if isinstance(name_or_num, int):
+            return self._bits[name_or_num].mask
+        else:
+            mask = 0
+            for name in name_or_num.split('|'):
+                mask |= self._bits[name].mask
+            return mask
+
+    def names(self, mask=None):
+        """Return list of names of masked bits.
+        If mask=None, return names of all known bits.
+        """
+        names = list()
+        if mask is None:
+            #  return names in sorted order of bitnum
+            bitnums = [x for x in self._bits.keys() if isinstance(x, int)]
+            for bitnum in sorted(bitnums):
+                names.append(self._bits[bitnum].name)
+        else:
+            bitnum = 0
+            while bitnum**2 <= mask:
+                if (2**bitnum & mask):
+                    if bitnum in self._bits.keys():
+                        names.append(self._bits[bitnum].name)
+                    else:
+                        names.append('UNKNOWN' + str(bitnum))
+                bitnum += 1
+
+        return names
+
+    #  Allow access via mask.BITNAME
+    def __getattr__(self, name):
+        if name in self._bits:
+            return self._bits[name]
+        else:
+            raise AttributeError('Unknown mask bit name ' + name)
+
+    #  What to print
+    def __repr__(self):
+        result = list()
+        result.append(self._name + ':')
+        #  return names in sorted order of bitnum
+        bitnums = [x for x in self._bits.keys() if isinstance(x, int)]
+        for bitnum in sorted(bitnums):
+            bit = self._bits[bitnum]
+            result.append('    - [{:16s} {:2d}, "{}"]'.format(
```

Comment:
Is there a particular reason the name field has to be 16 characters wide?  The problem is when you add in the missing comma, the output looks a bit odd:
```
ccdmask:
    - [BAD             ,  0, "Pre-determined bad pixel (any reason)"]
    - [HOT             ,  1, "Hot pixel"]
    - [DEAD            ,  2, "Dead pixel"]
    - [SATURATED       ,  3, "Saturated pixel from object"]
    - [COSMIC          ,  4, "Cosmic ray"]
```